infra/build/developer-tools/build/scripts/task_helper_functions.sh

```diff
@@ -384,7 +384,22 @@ function check_headers() {
   echo "Checking file headers"
   # Use the exclusion behavior of find_files
   find_files . -type f -print0 \
-    | compat_xargs -0 python /usr/local/verify_boilerplate/verify_boilerplate.py
+    | compat_xargs -0 python /usr/local/verify_boilerplate/verify_boilerplate.py 2>&1 \
+    | grep -v 'have incorrect boilerplate headers' > legacy_check.log
+  if [[ -v EXCLUDE_HEADER_CHECK ]]; then
+    SKIP_STRING=""
+    for file in ${EXCLUDE_HEADER_CHECK}
+    do
+      SKIP_STRING="$SKIP_STRING -skip $file"
+    done
+    stored_cmd="addlicense -check $SKIP_STRING ./ 2>&1 | grep -v 'skipping this file'"
+    eval "$stored_cmd" > new_check.log
+  else
+    addlicense -check ./ 2>&1 > new_check.log
+  fi
+  # list only the files caught by both methods, if not assume success header check
```

Comment:
What's the reason for checking both? Are they not compatible?

Reply:
Yes, as discussed in this [thread](https://chat.google.com/room/AAAAWaRnhC8/23oJnRQ8Zv8) there are cases where the `legacy_check` fails when the `addlicense_check` passes. It might be the case the other way round too. So as guided in [the thread](https://chat.google.com/room/AAAAWaRnhC8/23oJnRQ8Zv8) I am trying to make sure, previous CI checks that were passing with just the `legacy_check` continue to pass. 

Another way would be to advertise migration to addlicense and ask everyone to fix the headers by runnign the addlicense tool going forward.

Reply:
Got it. Can you have it log cases where addlicense fails with a warning? I'd like to diagnose those / phase out the legacy check over time.

Reply:
Sure. I will add the logs for this

Reply:
@Shabirmean I think it could be useful to have a `fix_headers` fn exposed which targets the same files to help ease the migration and ongoing maintenance.

Reply:
@bharathkkb - sure. I can add that.

@morgante - what is logs you expect?
- files that pass legacy check but fail the addlicense one? *(this I haven't encountered)*
- files that pass the addlicense but fail the legacy check? *(this is the one that I have encountered)*

Reply:
@Shabirmean I was hoping for logs of "files that pass legacy check but fail the addlicense one."

If you haven't found this so far, I think we should just remove the legacy check entirely. It sounds like `addlicense` is backwards-compatible with our existing headers.

Reply:
Added new method called `fix_headers`
Moved exclusion logic to `find_files`
Removed all legacy header checks and kept only `addlicence`